### PR TITLE
QUARKUS-1676 mTLS with OIDC Server

### PR DESCRIPTION
**Pull request by michalvavrik**
Provides a test plan for [QUARKUS-1676](https://issues.redhat.com/browse/QUARKUS-1676)

### CODE REVIEW

File: QUARKUS-1676.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# QUARKUS-1676 - Add support for mTLS with OIDC Server
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1676
+
+PR: https://github.com/quarkusio/quarkus/pull/23145
+
+Solving issue: https://github.com/quarkusio/quarkus/issues/19634#issuecomment-999435505
```
Comment (mjurc) [inline QUARKUS-1676.md:7]: Maybe 'Upstream issue' would sound better.

File: QUARKUS-1676.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Feature enables an OIDC client to present its TLS certificate when required by an OIDC server and thus have both parties
```
Comment (mjurc) [inline QUARKUS-1676.md:9]: This feature enables...

File: QUARKUS-1676.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+- Ensure documentation provides clear explanation on configuration options
+- Ensure documentation uses standard terminology and config options are relatable to Ver.x core options used underneath
+- Understand limitations of implementation
+- Check keystore types (namely JKS, PKCS12; other proprietary keystore types should be skipped)
+- Ensure good user experience and simplicity of use
+
+## Advanced topics for test development
+- Dev Services For Keycloak currently (Quarkus 2.8.0. Final) don't support mTLS (as requiring client
```
Comment (mjurc) [inline QUARKUS-1676.md:40]: This should maybe have its own issue linked to RFE in Jira, for developer experience

File: QUARKUS-1676.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  auth is build option, and you also can't provide your keystore).
```
Comment (mjurc) [inline QUARKUS-1676.md:41]: Missing contacts and references section

**Review by mjurc — CHANGES_REQUESTED** [review]
Thanks for the PR.

Also, please squash commits - there should be one commit for one logical unit of work.

File: QUARKUS-1676.md (hunk 1 above)
Comment (michalvavrik) [reply QUARKUS-1676.md]: Fixed, thanks.

File: QUARKUS-1676.md (hunk 2 above)
Comment (michalvavrik) [reply QUARKUS-1676.md]: Fixed, thanks.

File: QUARKUS-1676.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+- Keycloak and RH SSO (latest and stable version).
+
+### Impact on testsuites and testing automation:
+- Ensure this coverage works fine on DEV mode
+- Ensure this coverage works fine on Unix and Windows environments (for coverage that does not need Docker as this is a limitation in Windows)
+
+## Getting familiar with the feature
+Following actions were taken to ensure familiarity:
+- Ensure documentation provides clear explanation on configuration options
+- Ensure documentation uses standard terminology and config options are relatable to Ver.x core options used underneath
```
[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-1676.md]: Fixed, thanks.

File: QUARKUS-1676.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+## Scope of the testing
+
+Tests should verify:
+- User is able to add certificates to keystore and truststore to app run in a Java mode and a Native mode
+- Certificates and all relevant configurable options are picked up and used correctly by OIDC client
+- OIDC client and server communication is secured by mTLS when OIDC server requires so
+- Keycloak and RH SSO (latest and stable version).
+
+### Impact on testsuites and testing automation:
```
[the remark this answers is not in the document]
Comment (michalvavrik) [reply QUARKUS-1676.md:27]: Fixed, thanks.

File: QUARKUS-1676.md (hunk 4 above)
Comment (michalvavrik) [reply QUARKUS-1676.md]: Fixed, thanks.

**Comment by michalvavrik** [comment]
Thanks for a hint, commits squashed.

File: QUARKUS-1676.md (hunk 3 above)
Comment (michalvavrik) [reply QUARKUS-1676.md]: I described better issue with "Dev Services for Keycloak", however I haven't filled an RFE yet. I am honestly not sure if the issue is big enough, it looks to me like whoever wants to use mTLS with Dev Services for Keycloak can simply build his own image (especially as you always want to run your development against Keycloak version used in production, so you have to set image version manually anyway).

However I am happy to create a ticket if you think it's convenient!

Michal

File: QUARKUS-1676.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+New scenarios are going to be added to Quarkus Test suite with relatively low impact. Each test will
+require its own @QuarkusApplication as communication with OIDC server/setting of a keystore/truststore
+must be done during a build time.
+
+### Impact on resources:
+
+- No additional requirements for resources in lab
+- Overall Quarkus Test suite execution will be prolonged due to the OIDC server container startup time.
```
Comment (mjurc) [inline QUARKUS-1676.md:36]: Can we have information about order of the time required for the execution?

**Review by mjurc — CHANGES_REQUESTED** [review]

File: QUARKUS-1676.md (hunk 7 above)
Comment (michalvavrik) [reply QUARKUS-1676.md]: Information added.

**Comment by michalvavrik** [comment]
I enhanced information on resources requirements.

**Review by mjurc — APPROVED** [review]
LGTM!

**Pull request merged**